Client/Frontend/Widgets/ThumbnailCell.swift

```diff
@@ -36,47 +46,101 @@ class ThumbnailCell: UICollectionViewCell {
+        return UIVisualEffectView(effect: blur)
+    }()
+
+    lazy var imageWrapper: UIView = {
+        let imageWrapper = UIView()
         imageWrapper.layer.borderColor = ThumbnailCellUX.BorderColor.CGColor
-        imageWrapper.layer.borderWidth = 1
-        imageWrapper.layer.cornerRadius = 3
+        imageWrapper.layer.borderWidth = ThumbnailCellUX.BorderWidth
+        imageWrapper.layer.cornerRadius = ThumbnailCellUX.CornerRadius
         imageWrapper.clipsToBounds = true
+        return imageWrapper
+    } ()
 
+    lazy var removeButton: UIButton = {
+        let removeButton = UIButton()
         removeButton.setImage(UIImage(named: "TileCloseButton"), forState: UIControlState.Normal)
         removeButton.addTarget(self, action: "SELdidRemove", forControlEvents: UIControlEvents.TouchUpInside)
         removeButton.hidden = true
         removeButton.imageEdgeInsets = ThumbnailCellUX.RemoveButtonInsets
+        return removeButton
+    } ()
+
+    override init(frame: CGRect) {
+        imagePadding = CGFloat(0)
+        super.init(frame: frame)
+
+        isAccessibilityElement = true
+        addGestureRecognizer(longPressGesture)
+
+        contentView.addSubview(imageWrapper)
+        imageWrapper.addSubview(backgroundImage)
+        imageWrapper.addSubview(backgroundEffect)
+        imageWrapper.addSubview(imageView)
+        imageWrapper.addSubview(textWrapper)
+        textWrapper.addSubview(textLabel)
+        contentView.addSubview(removeButton)
 
         imageWrapper.snp_remakeConstraints({ make in
-            make.top.left.right.equalTo(self.contentView).insets(ThumbnailCellUX.Insets)
-            make.width.equalTo(self.imageWrapper.snp_height).multipliedBy(ThumbnailCellUX.ImageAspectRatio)
+            make.top.bottom.left.right.equalTo(self.contentView).insets(ThumbnailCellUX.Insets)
         })
 
-        imageView.snp_remakeConstraints({ make in
+        backgroundImage.snp_remakeConstraints({ make in
             make.top.bottom.left.right.equalTo(self.imageWrapper)
         })
 
-        textLabel.setContentHuggingPriority(1000, forAxis: UILayoutConstraintAxis.Vertical)
-        textLabel.font = ThumbnailCellUX.LabelFont
-        textLabel.textColor = ThumbnailCellUX.LabelColor
+        backgroundEffect.snp_remakeConstraints({ make in
+            make.top.bottom.left.right.equalTo(self.imageWrapper)
+        })
+
+        imageView.snp_remakeConstraints({ make in
+            let imagePadding: CGFloat = (UIScreen.mainScreen().traitCollection.horizontalSizeClass == .Compact) ? ThumbnailCellUX.ImagePaddingCompact : ThumbnailCellUX.ImagePadding
+            let insets = UIEdgeInsetsMake(imagePadding, imagePadding, imagePadding, imagePadding)
+            make.top.left.right.equalTo(self.imageWrapper).insets(insets)
+            make.bottom.equalTo(textWrapper.snp_top).offset(-imagePadding) // .insets(insets)
+        })
+
+        textWrapper.snp_remakeConstraints({ make in
```

Comment:
Nit: Instead of using snp_remakeConstraints, you can call snp_makeConstraints since you're not remaking them at this point